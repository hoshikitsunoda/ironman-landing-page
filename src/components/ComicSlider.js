import React, { Component } from 'react'

class ComicSlider extends Component {
  state = { comicData: [] }
  render() {
    return (
      <div>
        <div>ComicSlider</div>
      </div>
    )
  }
}

export default ComicSlider
