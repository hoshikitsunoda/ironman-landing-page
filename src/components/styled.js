import { createGlobalStyle } from 'styled-components'

// Global

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito|Ropa+Sans&display=swap');

  html {
    background-color: #fffafa;
  }
  
  html * {
    font-family: 'Nunito', sans-serif;
  }

  h1,
  h2,
  h3,
  h4,
  p {
    margin: 0 auto;
  }

  h1,
  h2,
  h3,
  h4 {
    font-family: 'Ropa Sans', sans-serif;
  }

  img {
    display: block;
  }
`
