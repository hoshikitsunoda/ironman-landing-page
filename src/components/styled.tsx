import { createGlobalStyle } from 'styled-components'

// Global

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito|Ropa+Sans&display=swap');

  html {
    background-color: #ffffff;
  }
  
  html * {
    font-family: 'Nunito', sans-serif;
  }

  body {
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  h1,
  h2,
  h3,
  h4,
  p {
    margin: 0 auto;
  }

  h1,
  h2,
  h3,
  h4 {
    font-family: 'Ropa Sans', sans-serif;
  }

  img {
    display: block;
  }
`

const size = {
  mobile: '350px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1200px',
}

export const Device = {
  mobile: `(min-width: ${size.mobile})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
}
