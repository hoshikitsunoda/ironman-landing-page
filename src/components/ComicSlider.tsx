import React from 'react'
import Slider from 'react-slick'
import { Comics, ComicsDataProps, SliderSetting } from '../models/ComicSlider'

import styled from 'styled-components'

const ComicSlider: React.FC<ComicsDataProps> = (props: ComicsDataProps) => {
  const settings: SliderSetting = {
    dots: false,
    infinite: true,
    slidesToShow: 6.5,
    slidesToScroll: 3,
    adaptiveHeight: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        },
      },
    ],
  }

  return (
    <StyledSlider {...settings}>
      {props.comicsData.map((item, i: number) => {
        const comic = new Comics(
          `${item && item.images[0].path}.${item && item.images[0].extension}`,
          item.urls[0].url,
          item.title
        )

        return (
          <div key={i}>
            <a href={comic.comicLink} rel="noopener noreferrer" target="_blank">
              <img src={comic.comicImage} alt={`${comic.comicTitle}`} />
            </a>
          </div>
        )
      })}
    </StyledSlider>
  )
}

const StyledSlider = styled(Slider)`
  .slick-list {
    padding: 2rem 0 2rem;
    margin-left: -3rem;

    .slick-slide {
      margin: 0 1.5rem 0 0;

      img {
        width: 100%;
      }
    }

    @media (min-width: 768px) {
      padding: 7rem 0 4rem;
      margin-left: -2rem;
    }
  }
`

export default ComicSlider
