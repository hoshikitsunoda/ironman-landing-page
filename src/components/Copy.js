import React from 'react'

import styled from 'styled-components'

const Copy = props => {
  const imgSrc = 'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55.jpg'
  return (
    <CopyWrapper>
      <HeadingWrapper className="text-wrapper heading">
        <p>Contrary to popular belief, he knows exactly what heâ€™s doing.</p>
      </HeadingWrapper>
      <div>
        <img src={imgSrc} alt="thumbnail" />
      </div>
    </CopyWrapper>
  )
}

const CopyWrapper = styled.div`
  padding: 0 1.2rem;

  img {
    width: 100%;
  }
`

const HeadingWrapper = styled.div`
  text-align: center;
  padding: 3rem 2.5rem;

  p {
    font-size: 1.5rem;
    line-height: 2.2rem;
  }
`

export default Copy
