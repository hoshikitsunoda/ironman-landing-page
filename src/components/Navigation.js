import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import Logo from './Logo'

const Navigation = () => {
  return (
    <NavWrapper className="nav-wrapper">
      <div className="logo">
        <Logo />
      </div>
      <div className="linkWrapper">
        <Link to="/">Sign In</Link>
        <Link to="/">Sign Up</Link>
      </div>
    </NavWrapper>
  )
}

const NavWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export default Navigation
