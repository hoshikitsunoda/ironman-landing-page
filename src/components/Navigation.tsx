import React from 'react'
import { BrowserRouter as Router, Link } from 'react-router-dom'

import styled from 'styled-components'
import * as Styled from '../components/styled'

import Logo from './Logo'

const Navigation: React.FC = () => {
  return (
    <NavWrapper className="nav-wrapper">
      <LogoWrapper className="logo-wrapper">
        <Logo />
      </LogoWrapper>
      <LinkWrapper className="linkWrapper">
        <Router>
          <NavLink to="/signin">Sign In</NavLink>
          <Button to="/signup">Sign Up</Button>
        </Router>
      </LinkWrapper>
    </NavWrapper>
  )
}

const NavWrapper = styled('div')`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 1.5rem 0;

  @media ${Styled.Device.tablet} {
    padding: 2.5rem 0;
  }
`

const LogoWrapper = styled('div')`
  width: 10.5rem;

  svg {
    width: 100%;
    height: auto;
    filter: brightness(0) saturate(100%) invert(1);
  }

  @media ${Styled.Device.tablet} {
    width: auto;
  }
`

const LinkWrapper = styled('div')`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: calc(100% - 6.5rem);

  @media ${Styled.Device.tablet} {
    width: auto;
  }

  @media ${Styled.Device.mobile} {
    width: calc(100% - 10.5rem);
  }
`

const Button = styled(Link)`
  background-color: #f4bc00;
  color: #fff;
  text-decoration: none;
  border-radius: 0.2rem;
  padding: 0.2rem 0.7rem;
  font-size: 0.75rem;
  cursor: pointer;
  box-shadow: 0 2px 5px -2px #fff;

  @media ${Styled.Device.tablet} {
    padding: 0.6rem 2rem;
    font-size: 1rem;
    font-weight: 700;

    &:hover {
      background-image: radial-gradient(
        70% 120%,
        #f4bc00 6%,
        #f4bc00 1%,
        #cc3332 79%
      );
      border-radius: 5px;
    }
  }
`

const NavLink = styled(Link)`
  color: #fff;
  text-decoration: none;
  margin: 0.2rem 0.7rem;
  font-size: 0.75rem;
  cursor: pointer;

  @media ${Styled.Device.tablet} {
    margin: 0.6rem 2rem;
    font-size: 1rem;
  }
`

export default Navigation
