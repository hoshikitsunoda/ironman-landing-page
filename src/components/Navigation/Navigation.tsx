import React from 'react'
import { BrowserRouter as Router, Link } from 'react-router-dom'

import styled from 'styled-components'
import * as Styled from '../../components/styled'

import Logo from '../Logo'

const Navigation: React.FC = () => {
  return (
    <NavWrapper className="nav-wrapper">
      <LogoWrapper className="logo-wrapper">
        <Logo />
      </LogoWrapper>
      <LinkWrapper className="link-wrapper">
        <Router>
          <NavLink to="/signin">Sign In</NavLink>
          <Button to="/signup">Sign Up</Button>
        </Router>
      </LinkWrapper>
    </NavWrapper>
  )
}

const NavWrapper = styled('div')`
  ${Styled.flexbox('row', 'space-between', 'flex-start')}
  padding: 1.5rem 0;

  @media ${(props) => props.theme.device.tablet} {
    padding: 2.5rem 0;
  }
`

const LogoWrapper = styled('div')`
  width: 10.5rem;

  svg {
    width: 100%;
    height: auto;
    filter: brightness(0) saturate(100%) invert(1);
  }

  @media ${(props) => props.theme.device.tablet} {
    width: auto;
  }
`

const LinkWrapper = styled('div')`
  ${Styled.flexbox('row', 'flex-end', 'center')}
  width: calc(100% - 6.5rem);

  @media ${(props) => props.theme.device.tablet} {
    width: auto;
  }

  @media ${(props) => props.theme.device.mobile} {
    width: calc(100% - 10.5rem);
  }
`

const Button = styled(Link)`
  background-color: ${(props) => props.theme.colors.primary};
  color: ${(props) => props.theme.colors.text};
  text-decoration: none;
  border-radius: 0.2rem;
  padding: 0.2rem 0.7rem;
  font-size: 0.75rem;
  cursor: pointer;
  box-shadow: 0 2px 5px -2px ${(props) => props.theme.colors.text};

  @media ${(props) => props.theme.device.tablet} {
    padding: 0.6rem 2rem;
    font-size: 1rem;
    font-weight: 700;

    &:hover {
      ${Styled.buttonHover}
      border-radius: 5px;
    }
  }
`

const NavLink = styled(Link)`
  color: ${(props) => props.theme.colors.text};
  text-decoration: none;
  margin: 0.2rem 0.7rem;
  font-size: 0.75rem;
  cursor: pointer;

  @media ${(props) => props.theme.device.tablet} {
    margin: 0.6rem 2rem;
    font-size: 1rem;
  }
`

export default Navigation
