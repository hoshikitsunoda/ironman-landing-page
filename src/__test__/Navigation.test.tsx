import React from 'react'
import { mount } from 'enzyme'
import Navigation from '../components/Navigation'
import Logo from '../components/Logo'

import { ThemeProvider } from 'styled-components'
import * as Styled from '../components/styled'

describe('<Navigation />', () => {
  const wrapper = mount(
    <ThemeProvider theme={Styled.theme}>
      <Navigation />
    </ThemeProvider>
  )
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Logo', () => {
    expect(wrapper.find(Logo)).toHaveLength(1)
  })

  it('should render two Links', () => {
    expect(wrapper.find('Link')).toHaveLength(2)
  })

  it('should have corresponding href', () => {
    expect(wrapper.find('Link').first().props().to).toEqual('/signin')
    expect(wrapper.find('Link').last().props().to).toEqual('/signup')
  })
})
