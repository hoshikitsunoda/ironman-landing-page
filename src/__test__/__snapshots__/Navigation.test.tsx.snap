// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Renders Navigation 1`] = `ShallowWrapper {}`;
